### PR TITLE
Test wiki snippets in CI

### PR DESCRIPTION
**Pull request by bobismijnnaam**
This PR adds functionality of testing all snippets present in the wiki when the wiki is updated, as well as including the snippets from the wiki in the regular test suite.

The wiki script is extended to dump all extracted cases to a folder with proper testing headers.

Finally, a github action is added which ensures in PRs the testing workflow is not run in duplo.

### CODE REVIEW

File: .github/workflows/build-test.yml [hunk 1]
```diff
@@ -10,7 +10,28 @@ on:
...
+    runs-on: ubuntu-latest
+    # Map a step output to a job output
+    outputs:
+      should_skip: ${{ steps.skip_check.outputs.should_skip }}
+    steps:
+      - id: skip_check
+        uses: fkirc/skip-duplicate-actions@master
+        with:
+          # Never allow two runs of this workflow to exist
+          concurrent_skipping: 'always'
```
Comment (niomaster) [inline .github/workflows/build-test.yml:23]: Does this skip the prior run or the latter?

File: .github/workflows/build-test.yml [hunk 2]
```diff
@@ -202,11 +223,44 @@ jobs:
...
+      - run: "sudo dpkg -i *.deb"
+      - run: vercors --silicon examples/manual/fibonacci.pvl
+      - run: vercors --test=wiki_cases --test-workers=2 --tool carbon --tool silicon --tool veymont --exclude-suite=slow,medium,problem-fail --progress --actions-test-output --enable-test-coverage --coverage-output-file=jacoco_wiki_tests.xml
+      - name: Archive Java Code Coverage (jacoco_wiki_tests.xml)
+        uses: actions/upload-artifact@v2
+        with:
+          name: vercors-jacoco-xml
+          retention-days: 1
+          path: |
+            jacoco*.xml
```
Comment (niomaster) [inline .github/workflows/build-test.yml:257]: Perhaps this should run without coverage?

Comment (niomaster) [reply .github/workflows/build-test.yml:257]: Perhaps it's also advisable to make a warning that changes to this sequence should be synchronized with the one in wiki-snippet-test

File: util/wiki/examplesToBeFixed.txt [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+specification-syntax-auxiliary-annotations-ghost-code-ghost-methods-and-functions-abstract-methods-and-functions-1-silicon
+specification-syntax-auxiliary-annotations-ghost-code-ghost-parameters-and-results-1-silicon
+specification-syntax-method-contracts-1-silicon
+axiomatic-data-types-option-2-silicon 
+
+To be fixed:
+gpgpu-verification-?-openclcuda-with-barrier-1-silicon
+gpgpu-verification-?-openclcuda-without-barrier-1-silicon
+gpgpu-verification-?-openclcuda-with-barrier-2-silicon
+gpgpu-verification-?-openclcuda-without-barrier-2-silicon
```
Comment (niomaster) [inline util/wiki/examplesToBeFixed.txt:28]: remove please

File: util/wiki/magicWand.java [hunk 4]
```diff
@@ -0,0 +1,113 @@
...
+    Tree left = get_left_sorted(t) /*@ then { root_prio = root_prio; } @*/;
+    // now we have left.tree(), sorted(left) and wand_helper(left, root_prio), as well as the wand
+    //@ unfold left.tree();
+    boolean res = left.priority <= max_priority;
+    //@ fold left.tree();
+    //@ apply (left.tree() ** sorted(left) ** wand_helper(left, root_prio)) -* (t.tree() ** sorted(t));
+    // now left.tree() is no longer directly available, but t.tree() is back and sorted
+    return res;
+  }
+}
```
Comment (niomaster) [inline util/wiki/magicWand.java:113]: hmm?

**Review by niomaster — CHANGES_REQUESTED** [review]

File: .github/workflows/build-test.yml (hunk 1 above)
Comment (bobismijnnaam) [reply .github/workflows/build-test.yml:23]: This is not clear to me from the docs, so I'll have to experiment

File: util/wiki/magicWand.java (hunk 4 above)
Comment (bobismijnnaam) [reply util/wiki/magicWand.java]: Removed it, or was there something that surprised you?

File: .github/workflows/build-test.yml (hunk 2 above)
Comment (bobismijnnaam) [reply .github/workflows/build-test.yml:258]: Hmm, I guess in that case the example/test suite should be improved to also cover that part, after which the coverage from the wiki is no longer relevant, so indeed you are right! I'll remove it.

**Pull request merged**